### PR TITLE
[#1526]: Preventing the problem of catching an empty list of signals

### LINKED ISSUE #1526: trio.open_signal_receiver should give an error if no arguments are passed
User confusion spotted in the wild: https://gitter.im/python-trio/general?at=5ebfa9d613878c30b581b9fe

### PR DESCRIPTION
**Pull request by kaldown**
closes #1526

### CODE REVIEW

File: trio/tests/test_signals.py [hunk 1]
```diff
@@ -41,6 +41,12 @@ async def test_open_signal_receiver_restore_handler_after_one_bad_signal():
     assert signal.getsignal(signal.SIGILL) is orig
 
 
+async def test_open_signal_receiver_empty_fail():
+    with pytest.raises(AttributeError):
```
Comment (pquentin) [inline trio/tests/test_signals.py:45]: You can pass `match=...` to `pytest.raises` to make sure you get the expected message. It's not mandatory or anything though

File: trio/_signals.py [hunk 2]
```diff
@@ -144,6 +144,9 @@ def open_signal_receiver(*signals):
                  reload_configuration()
 
     """
+    if not signals:
+        raise AttributeError("No signals were provided")
```
Comment (pquentin) [inline trio/_signals.py:148]: AttributeError is reserved to attributes of objects, I think ValueError is more appropriate here.

Comment (njsmith) [reply trio/_signals.py]: Or maybe `TypeError`? That's what you get if you try to call a function with too few mandatory arguments:

```
In [1]: def f(x): 
   ...:     pass 
   ...:                                                                                   

In [2]: f()                                                                               
TypeError: f() missing 1 required positional argument: 'x'
```

We should also add a newsfragment.

Comment (kaldown) [reply trio/_signals.py]: What's that 'newsfragment' ?

Comment (kaldown) [reply trio/_signals.py]: changed to `TypeError`

Comment (pquentin) [reply trio/_signals.py]: See https://trio.readthedocs.io/en/stable/contributing.html#pull-request-release-notes for details

**Comment by kaldown** [comment]
Finally, I won that `*.rst`.Tthis `__` at the end of the url, looks weird, but helpful.

File: newsfragments/1526.feature.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Some users are confused with the usage of `open_signal_receiver <https://trio.readthedocs.io/en/stable/reference-io.html#trio.open_signal_receiver>`__ being able to operate without any signal.
```
Comment (oremanj) [inline newsfragments/1526.feature.rst:1]: Just say
```
`open_signal_receiver`
```
and Sphinx will resolve the link for you automatically.

File: newsfragments/1526.feature.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
...
+This issue being fixed.
```
Comment (oremanj) [inline newsfragments/1526.feature.rst:2]: The wording you have here is not clear about what changed, which is the main point of a newsfragment. I would say something like: Calling `open_signal_receiver` with no arguments used to succeed without listening for any signals. This was confusing, so now it raises `ValueError` instead.

Also this isn't really a "feature" - I would probably put this in the "bug fixes" section of the release notes (so name the newsfragment "1526.bugfix.rst").

**Comment by oremanj** [comment]
Also, please write "Closes #1526" in the PR description so that merging the PR will close the issue. Just listing the issue number isn't sufficient for that.

File: newsfragments/1526.feature.rst (hunk 3 above)
Comment (kaldown) [reply newsfragments/1526.feature.rst:1]: I followed this example https://raw.githubusercontent.com/python-trio/trio/master/newsfragments/1241.feature.rst

Comment (pquentin) [reply newsfragments/1526.feature.rst]: (It also makes sure the function actually exists, which is handy to avoid dead links.)

**Comment by kaldown** [comment]
Thank you for the patient :)

**Pull request merged**